carapace-server/src/main/java/org/carapaceproxy/configstore/HerdDBConfigurationStore.java

```diff
@@ -310,19 +329,26 @@ private KeyPair loadKeyPair(String pk) throws Exception {
         }
     }
 
-    private void saveKeyPair(KeyPair pair, String pk) throws Exception {
+    private boolean saveKeyPair(KeyPair pair, String pk, boolean update) {
         try (Connection con = datasource.getConnection(); PreparedStatement psInsert = con.prepareStatement(INSERT_INTO_KEYPAIR_TABLE); PreparedStatement psUpdate = con.prepareStatement(UPDATE_KEYPAIR_TABLE)) {
             String privateKey = base64EncodeKey(pair.getPrivate());
             String publicKey = base64EncodeKey(pair.getPublic());
-            psUpdate.setString(1, privateKey);
-            psUpdate.setString(2, publicKey);
-            psUpdate.setString(3, pk);
-            if (psUpdate.executeUpdate() == 0) {
+            boolean updateDone = false;
+            if (update) {
```

Comment:
what does 'update = false' mean ?
this method won't do anything

Reply:
Maybe it's safer to allow only KeyPairs inserts?

Reply:
Yes, probably